### ISSUE TITLE
Support casting VARCHAR to XML / XMLTYPE in Oracle

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13255]
This isn't valid in Oracle:

```sql
select cast('<a/>' as xmltype) from dual;
```

But this is:

```sql
select xmltype('<a/>') from dual;
```

Our `CAST` expression should implement the above behaviour

**Comment by lukaseder** [comment]
This works correctly in Db2 and PostgreSQL, so it's an Oracle-only issue.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0

**Comment by lukaseder** [comment]
Would be worth parsing `XMLTYPE(...)` into a `CAST` expression, too
